atomic_reactor/utils/koji.py

```diff
@@ -400,25 +407,24 @@ def get_maven_metadata(workflow_data):
     components = fetch_maven_results.get('components', [])
 
     for remote_source_file in maven_url_sources_metadata_results.get('remote_source_files', []):
-        outputs.append(Output(file=open(remote_source_file['file']),
+        outputs.append(Output(filename=remote_source_file['file'],
                               metadata=remote_source_file['metadata']))
     return outputs, components
 
 
-def get_image_components(workflow):
+def get_image_components(workflow, platform: str) -> List[Dict[str, Union[str, int]]]:
     """
     Re-package the output of the rpmqa plugin into the format required
     for the metadata.
     """
-    output = workflow.data.image_components
-    if output is None:
+    components = workflow.data.postbuild_results.get(PostBuildRPMqaPlugin.key)
+    if components is None:
```

Comment:
I don't think the result of `rpmqa` will ever be `None`, at most it'll be something likes `{platfrom: None}` since we are using `build_dir.for_each_platfrom` to return the results

Reply:
Hi @hjmodi, thanks for your comment. As you can see the following `logger.error`, this check is checking whether `PostBuildRPMqaPlugin` plugin has run. If not, no result in the `postbuild_results`.

Reply:
hmm, I do understand that. It'll only not run for source build in which case this function will not be called. So I don't think we need to check for that. It might lead to confusion. It'll always run for the binary build, if no RPMs are found it'll return a `{plafrom: None}` dictionary

Reply:
My understand, the original decision of this check is based on the plugins relationship rather than what kind of data is set to the `image_components`. For now, I'd like keep this original behavior unchanged. Maybe, `log.warning` is more proper than `log.error`.